frappe/public/js/frappe/form/form.js

```diff
@@ -1658,6 +1658,21 @@ frappe.ui.form.Form = class FrappeForm {
 		frappe.route.on('change', () => driver.reset());
 		driver.start();
 	}
+
+	// Filters fields from the reference doctype and sets them as options for a Select field
+	set_fields_as_options(fieldname, reference_doctype, filter_function, default_options=[], table_fieldname) {
+		if (!reference_doctype) return;
+		let options = default_options;
+		return new Promise(resolve => {
+			frappe.model.with_doctype(reference_doctype, () => {
+				frappe.get_meta(reference_doctype).fields.map(df => {
+					filter_function(df) && options.push({ label: df.label, value: df.fieldname });
+				});
+				options && this.set_df_property(fieldname, 'options', options, this.doc.name, table_fieldname);
+				resolve(options);
+			});
+		});
+	}
```

Comment:
@rmehta this function can be used to set fields as options.

Reply:
Cool thanks 👍